lib/plugins/manifest.js

```diff
@@ -33,7 +33,15 @@ module.exports = function(plugins, webpackConfig) {
             const isJsOrJsMapFile = /\.js(\.map)?$/.test(file.name);
 
             return !isCopyEntry && !(isStyleEntry && isJsOrJsMapFile);
-        }
+        },
+        map: (file) => {
+            if (webpackConfig.useVersioning) {
+                // Remove hash in manifest key
+                file.name = file.name.replace(/(\.[a-f0-9]{8})(\..*)$/, '$2');
```

Comment:
Hi @shadowc,

I'm not sure it would always work... IIRC the hard thing to handle here is that the user has control on the naming, so you can't really assume that:

* hashes will be 8 characters long (users could chose a different length)
* hashes will only be composed of hexadecimal characters (hash function can be changed as well)
* hashes will be at that exact position (eg. `[contenthash].[name].js`)
* original filenames won't contain something that looks like a hash (eg. `foo.aaaaaaaa.d1fc2a.txt` would become `foo.d1fc2a.txt`)

Reply:
Thanks for the heads up! We do have the info at that point of what format the user choose for the `to` file name, do we? Also, there is a default format `[name].[contenthash].[ext]` that encore uses?

I'll investigate this further and put against several scenarios to double check!

Reply:
To answer my own question, I believe we don't have all this information.

I will think of ways to cache regexp patterns for each `CopyWebpackPlugin` entry together with the position of the hash, then use this to remove such hash from the entry in this function. I believe this could work.

Reply:
On the other hand, what I need to cache is the name of the original entry file (not the `to` field) and just retrieve that somehow.

Reply:
Here's a simplified version of how this should work:

```javascript
static_file_mapping = {};
config = {
  plugins: [
    new CopyPlugin([{
            from: {
                glob: 'static/**/*.@(png|svg|jpg|gif|ico)',
                dot: false,
                noext: false,
            },
            to: '[contenthash].[ext]',
            // flatten: true
            transformPath(targetPath, absosutePath) {
                relative_path = path.relative(
                    path.resolve(root_dir),
                    absosutePath)
                static_file_mapping[relative_path] = targetPath;
                return targetPath;
            }
    }]),
    new ManifestPlugin({
      assets: static_file_mapping,
    })
  ]
}
```

Reply:
If that helps here is the PR in which I initially tried to implement `copyFiles()` using the WebpackManifestPlugin: https://github.com/symfony/webpack-encore/pull/221

As you will be able to read in the comments I couldn't make it work at that time because the CopyWebpackPlugin didn't expose original filenames. It looks like some work as been done since then: https://github.com/webpack-contrib/copy-webpack-plugin/pull/309#issuecomment-721095616

Not sure if that can be used without modifying the WebpackManifestPlugin internals though.

Reply:
Thank you! I'll look into this and come up with a better solution!